src/Contrib/Newrelic/SpanConverter.php

```diff
@@ -40,12 +35,16 @@ private function convertSpan(SpanDataInterface $span): array
         $startTimestamp = TimeUtil::nanosToMillis($span->getStartEpochNanos());
         $endTimestamp = TimeUtil::nanosToMillis($span->getEndEpochNanos());
 
+        $serviceName =  $span->getResource()->getAttributes()->get(ResourceAttributes::SERVICE_NAME)
+                        ??
+                        ResourceInfoFactory::defaultResource()->getAttributes()->get(ResourceAttributes::SERVICE_NAME);
```

Comment:
depending on how many detectors are enabled (by default, all of them are), this is a bit of overhead we could avoid? Perhaps setting a default/fallback service name in the constructor?

Reply:
Good point